tests/view/RoomsTest.java

```diff
@@ -0,0 +1,178 @@
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.swing.*;
+import okhttp3.OkHttpClient;
+import org.junit.Assert;
+import org.junit.Test;
+import use_case.search.SearchDataAccessInterface;
+
+public class RoomsTest extends ButtonTest {
+
+  @Test
+  public void testClickingRefresh() {
+    RoomsViewModel roomsViewModel = new RoomsViewModel();
+    OkHttpClient client = new OkHttpClient();
+    RoomsView roomsView =
+        RoomsUseCaseFactory.create(
+            new FirebaseRoomsDataAccessObject(client),
+            new FirebaseMessageDataAccessObject(client),
+            new FirebaseUserDataAccessObject(client),
+            roomsViewModel,
+            null,
+            null);
+
+    roomsViewModel.setState(this.buildTestState());
+
+    JButton refresh = roomsView.getRefreshButton();
+    refresh.doClick();
+
+    // User can load blank rooms with request to Firebase
+    Assert.assertTrue(roomsViewModel.getState().getError() == null);
+  }
+
+  @Test
+  public void testRoomCreation() {
+    RoomsViewModel roomsViewModel = new RoomsViewModel();
+    OkHttpClient client = new OkHttpClient();
+    RoomsView roomsView =
+        RoomsUseCaseFactory.create(
+            new FirebaseRoomsDataAccessObject(client),
+            new FirebaseMessageDataAccessObject(client),
+            new FirebaseUserDataAccessObject(client),
+            roomsViewModel,
+            null,
+            null);
+
+    RoomsState testState = buildTestState();
+    testState.setRoomToCreateName("Test Room!");
+    roomsViewModel.setState(testState);
+
+    JButton createRoomButton = roomsView.getCreateRoomButton();
+    createRoomButton.doClick();
+
+    String regex = "^Authentication failed: .*";
+
+    Pattern pattern = Pattern.compile(regex);
+    Matcher matcher = pattern.matcher(roomsViewModel.getState().getError());
+    Assert.assertTrue(matcher.find());
+  }
+
+  @Test
+  public void testSendMessage() {
+    OkHttpClient client = new OkHttpClient();
+    ViewManagerModel viewManagerModel = new ViewManagerModel();
+
+    SearchViewModel searchViewModel = new SearchViewModel();
+    SearchedViewModel searchedViewModel = new SearchedViewModel();
+    SearchDataAccessInterface searchDataAccessObject = new ElasticsearchDataAccessObject(client);
+    SearchController searchController =
+        SearchUseCaseFactory.createSearchController(
+            searchViewModel, searchDataAccessObject, viewManagerModel, searchedViewModel);
+
+    RoomsViewModel roomsViewModel = new RoomsViewModel();
+    RoomsView roomsView =
+        RoomsUseCaseFactory.create(
+            new FirebaseRoomsDataAccessObject(client),
+            new FirebaseMessageDataAccessObject(client),
+            new FirebaseUserDataAccessObject(client),
+            roomsViewModel,
+            searchController,
+            null);
```

Comment:
interesting and a bit worrying that this works, but ok

Reply:
?

Reply:
(Passing the null in instead of a controller)